crawler/priority_samples_process.py

```diff
@@ -0,0 +1,267 @@
+    IMPORTANT_UNPROCESSED_SAMPLES_MONGO_QUERY: Final[List[object]] = [
+        # first get only unprocessed priority samples
+        {
+            "$match": {
+                "$and": [
+                    {FIELD_PROCESSED: False},
+                ]
+            }
+        },
+        {
+            "$lookup": {
+                "from": "samples",
+                "localField": "sample_id",
+                "foreignField": FIELD_MONGODB_ID,
+                "as": "related_samples",
+            }
+        },
+        # replace the document with a merge of the original and the first element of the array created from the lookup
+        # above - this should always be 1 element
+        {"$replaceRoot": {"newRoot": {"$mergeObjects": [{"$arrayElemAt": ["$related_samples", 0]}, "$$ROOT"]}}},
+        # result = { mongo_sample_id=1234, must_sequence=true; related_samples: { uuid: 132 } }
+        # remove the lookup document
+        {"$project": {"related_samples": 0}}
+        # result = { mongo_sample_id=1234, must_sequence=true, uuid: 132 }
+    ]
+
+    value = priority_samples_collection.aggregate(IMPORTANT_UNPROCESSED_SAMPLES_MONGO_QUERY)
+    return list(value)
+
+
+def merge_priority_samples_into_docs_to_insert(priority_samples: List[Any], docs_to_insert: list) -> None:
+    """
+    Updates the sample records with must_sequence and preferentially_sequence values
+
+    for each successful add sample, merge into docs_to_insert_mlwh
+    with must_sequence and preferentially_sequence values
+
+    Arguments:
+        priority_samples  - priority samples to update docs_to_insert with
+        docs_to_insert {List[ModifiedRow]} -- the sample records to update
+    """
+
+    def extract_root_sample_id(sample: SampleDoc) -> SampleDocValue:
+        return sample[FIELD_ROOT_SAMPLE_ID]
+
+    priority_root_sample_ids = list(map(extract_root_sample_id, priority_samples))
+
+    for doc in docs_to_insert:
+        root_sample_id = doc[FIELD_ROOT_SAMPLE_ID]
+        if root_sample_id in priority_root_sample_ids:
+            priority_sample = list(filter(lambda x: x[FIELD_ROOT_SAMPLE_ID] == root_sample_id, priority_samples))[0]
+            doc[FIELD_MUST_SEQUENCE] = priority_sample[FIELD_MUST_SEQUENCE]
+            doc[FIELD_PREFERENTIALLY_SEQUENCE] = priority_sample[FIELD_PREFERENTIALLY_SEQUENCE]
+
+
+# TODO: refactor duplicated function
+def update_unprocessed_priority_samples_to_processed(db: Database, sample_ids: list) -> None:
+    """
+    Description
+    use stored identifiers to update priority_samples table to processed true
+    Arguments:
+        x {Type} -- description
+    """
+    priority_samples_collection = get_mongo_collection(db, COLLECTION_PRIORITY_SAMPLES)
+    for sample_id in sample_ids:
+        priority_samples_collection.update({FIELD_SAMPLE_ID: sample_id}, {"$set": {FIELD_PROCESSED: True}})
+    logger.info("Mongo update of processed for priority samples successful")
+
+
+# TODO: refactor duplicated function
```

Comment:
still duplicated?

Reply:
Yes - Duplicate method to insert_samples_from_docs_into_mlwh in file_processing.py/, possibly refactor to MLWH helper??

Reply:
update_priority_samples_into_mlwh / insert_samples_from_docs_into_mlwh -> Refactor after deploying to UAT